### ISSUE TITLE
Approved reviews by github-actions[bot] are not recognized by Mergify.io

### ISSUE THREAD

**Issue by allez-allez-allez** [Mergifyio/mergify-engine#514]
We are testing out Github Actions, and Mergify.io to automate Dependabots pull requests.  The flow is quite simple. Since our master branches are protected and required at least one reviewer, so these pull requests need to be approved first. When Dependabot creates a new pr, the `auto-approve-action` (https://github.com/hmarr/auto-approve-action) runs and approves the pr. After that mergify.io will merge the pull request into the master branch.

Configuration for the `auto-approve-action`
```yaml
name: auto approve Dependabot pull requests
on: pull_request

jobs:
  auto-approve-dependabot-pr:
    runs-on: ubuntu-latest
    steps:
      - name: auto approve dependabot pull requests
        uses: hmarr/auto-approve-action@master
        if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
        with:
          github-token: "${{ secrets.GITHUB_TOKEN }}"
```
Configuration for `mergify`
```yaml
pull_request_rules:
  - name: automatic merge for approved Dependabot pull requests
    conditions:
      - author=dependabot-preview[bot]
      - status-success=Linting - Check Style
      - "#approved-reviews-by>=1"
    actions:
      merge:
        method: merge
```

When `auto-approve-action` approves the pull request and uses `GITHUB_TOKEN`, GitHub creates an approved review with `github-actions[bot]` as the author (more information about `GITHUB_TOKEN` can be found [here](https://help.github.com/en/articles/virtual-environments-for-github-actions#github_token-secret). 

<img width="803" alt="Screen Shot 2019-10-09 at 8 50 32 AM" src="https://user-images.githubusercontent.com/48020826/66482874-eb378880-ea71-11e9-9ddb-f8ca0f495ac9.png">

Even though `github-action[bot]` has write permissions to the repo, Megify is not recognizing it as a legitimate user, therefore the condition `"#approved-reviews-by>=1"` is always failing. 

<img width="424" alt="Screen Shot 2019-10-09 at 8 51 34 AM" src="https://user-images.githubusercontent.com/48020826/66482975-1e7a1780-ea72-11e9-9f0e-e52060d29007.png">

Is there a configuration setting that we can specify to make mergify recognize `github-actions[bot]` as a user and count its approved review? if there are none, then is this something we can work on and then submit a pr to you guys to merge into the baseline? 

Thanks, love your product!

**Comment by sileht** [comment]
If you are using branch protection, you can safely remove `- "#approved-reviews-by>=1"`. When merge action is run, Mergify always wait the branch protection to be validated before merging the PR.

**Comment by allez-allez-allez** [comment]
tested and it worked. Thanks for the prompt reply.

**Comment by jd** [comment]
I guess `github-actions[bot]` is not recognized as a valid reviewer because it's not a member of the repository or the like. I'd be curious to know what is its status inside the repository.
